helper-cli/src/main/kotlin/common/DatabaseHelper.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.helper.common
+
+import org.jetbrains.exposed.sql.Database
+
+import org.ossreviewtoolkit.model.config.OrtConfiguration
+import org.ossreviewtoolkit.model.config.PostgresStorageConfiguration
+import org.ossreviewtoolkit.model.utils.DatabaseUtils
+import org.ossreviewtoolkit.scanner.LocalScanner
+import org.ossreviewtoolkit.utils.log
+
+internal object DatabaseHelper {
+    fun initDatabase(config: OrtConfiguration): Database {
+        val storageConfig = config.scanner.storages?.get("postgresStorage") as PostgresStorageConfiguration
+
+        log.info {
+            "Using Postgres storage with URL '${storageConfig.url}' and schema '${storageConfig.schema}'."
+        }
+
+        val dataSource = DatabaseUtils.createHikariDataSource(
+            config = storageConfig,
+            applicationNameSuffix = ORTH_NAME,
+            // Use a value slightly higher than the number of threads accessing the storage.
+            maxPoolSize = LocalScanner.NUM_STORAGE_THREADS + 3
```

Comment:
This is probably a copy & paste error. There is only a single thread accessing the database. So you could set `maxPoolSize` to a rather low value, maybe even 1.

Reply:
As this is implemented as a Utitlity class I think that it will be used for different use cases as well, which might access the database in multiple threads.
Do you think we'll remember to raise the pool size then?

However, you're right, this is a copy paste error, I'd suggest to just use the default pool size (5).
What do you think?

Reply:
Hikari will remind us to increase the pool size by spitting out strange errors related to timeouts ;-)

But kidding aside, if it is a generic utility function, then the pool size should be a parameter - as you said, it depends on the use case.

Reply:
I moved this functionality to a private method.
So it is no longer a utility function and I reduced the `maxPoolSize` to `1`.